src/augment-pr-entry/augment-pr-entry.js

```diff
@@ -10,8 +12,13 @@ import linkifyTargetBranch from '../linkify-target-branch/linkify-target-branch'
 
 const repoUrl = getRepoURL();
 
-export const getPrData = async prId => {
-    const url = `https://api.bitbucket.org/2.0/repositories/${repoUrl}/pullrequests/${prId}`;
+export const getPrData = async (prId, collection) => {
+    const baseUrl = `https://api.bitbucket.org/2.0/repositories/${repoUrl}/pullrequests/${prId}`;
+    const url =
+        typeof collection === 'undefined'
+            ? baseUrl
+            : `${baseUrl}/${collection}`;
+
```

Comment:
This function is now performing two separate type of requests, and one of them doesn't really match the name of the function. I think it would be best to split them up and have a common third one that performs the request, something like this:

```js
const request = async url => {
    // ...
};

export const getPrActivity = prId => {
    const url =
        `https://api.bitbucket.org/2.0/repositories/${repoUrl}/pullrequests/${prId}/activity` +
        '?pageLen=1';
    return request(url);
};

export const getPrData = prId => {
    const url = `https://api.bitbucket.org/2.0/repositories/${repoUrl}/pullrequests/${prId}`;
    return request(url);
};
```

Also notice how I appended the `pageLen` param to the URL in `getPrActivity` . We only actually need the latest activity item, so anything else than one entry is gonna be a waste of bandwith.

Reply:
Originally thought of that but the structure of the APIs `/pullrequests/{prId}/[comments|activity|commits`  looked convincing on reusing the the existing function. BitBucket's pagination querystring appears to be case-sensitive - so `pagelen` instead of `pageLen`.  Overall, yours is the better approach - single responsibility & reduced bandwidth  👍.